File: Files/MVP.xctemplate/___FILEBASENAME___/Presenter/___FILEBASENAME___Presenter.m
//
//  ___FILENAME___
//  ___PROJECTNAME___
//
//  Created by ___FULLUSERNAME___ on ___DATE___.
//
 
#import "___VARIABLE_mvpModuleName___Presenter.h"

@implementation ___VARIABLE_mvpModuleName___Presenter

- (instancetype)initWithInterface:(id<___VARIABLE_mvpModuleName___ViewProtocol>)interface router:(id<___VARIABLE_mvpModuleName___WireframeProtocol>)router
{
    if (self = [super init]) {
        self.view = interface;
        self.router = router;
    }
    return self;
}

- (void)configData { }

@end

File: Files/VIPER.xctemplate/___FILEBASENAME___/___FILEBASENAME___Presenter.swift
//
//  ___FILENAME___
//  ___PROJECTNAME___
//
//  Created ___FULLUSERNAME___ on ___DATE___.
//

import UIKit

final class ___VARIABLE_productName:identifier___Presenter: ___VARIABLE_productName:identifier___PresenterProtocol {

    weak private var view: ___VARIABLE_productName:identifier___ViewProtocol?
    var interactor: ___VARIABLE_productName:identifier___InteractorProtocol?
    private let router: ___VARIABLE_productName:identifier___WireframeProtocol

    init(interface: ___VARIABLE_productName:identifier___ViewProtocol, interactor: ___VARIABLE_productName:identifier___InteractorProtocol?, router: ___VARIABLE_productName:identifier___WireframeProtocol) {
        self.view = interface
        self.interactor = interactor
        self.router = router
    }

}
